src/puya/ir/context.py

```diff
@@ -63,7 +63,10 @@ def resolve_function_reference(
                             invocation.source_location,
                         )
                 case awst_nodes.FreeSubroutineTarget(module_name=module_name, name=func_name):
-                    func = self.module_awsts[module_name].symtable[func_name]
+                    if module_name == "_puyapy_":
```

Comment:
This is required if a subroutine in the `_puyapy_` module references another subroutine inside the `_puyapy_` module. 

After moving the internal helpers into their own module - this is no longer required but maybe it will be in the future?

Reply:
Yeah a bit of a quirk due to us having something in a puyapy stub and as an embedded implementation. Maybe leave it in, but add a comment to explain why it is here

Reply:
Added a comment while making other changes here